src/puppetlabs/trapperkeeper/internal.clj

```diff
@@ -44,6 +47,25 @@
    (every? keyword? (keys service-graph))
    (every? (some-fn ifn? service-graph?) (vals service-graph))))
 
+(defn inc-restart-counter!
+  "Increments the counter in a restart-file for purposes of supporting HUP behavior"
+  [restart-file-path]
+  (try
+    (if (fs/exists? restart-file-path)
+      (if (and (fs/writeable? restart-file-path) (fs/readable? restart-file-path))
+        (spit restart-file-path (inc (Long/parseLong (slurp restart-file-path))))
+        (log/error "Restart File is not readable and/or writeable"))
+      (let [dir-path (fs/parent restart-file-path)]
+        (fs/mkdirs dir-path)
+        (spit restart-file-path "0")
```

Comment:
but if you spit "1", you don't need to call `inc-restart-counter!` again, you can just exit the function.